consensus/spos/bls/subroundEndRound.go

```diff
@@ -108,6 +118,9 @@ func (sr *subroundEndRound) doEndRoundJob() bool {
 	msg := fmt.Sprintf("Added proposed block with nonce  %d  in blockchain", sr.Header.GetNonce())
 	log.Info(log.Headline(msg, sr.SyncTimer().FormattedCurrentTime(), "+"))
 
+	if sr.appStatusHandler != nil {
```

Comment:
move this check from here on SetAppStatusHandler func

Reply:
done